File: frontend/src/migrate_theme.js
const fs = require('fs');
const path = require('path');

function migrateFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf-8');
    const original = content;

    // Backgrounds
    content = content.replace(/bg-\[#0a0e1a\]/g, 'bg-background');
    content = content.replace(/bg-\[#0f1629\]/g, 'bg-background');
    content = content.replace(/bg-\[#111827\]/g, 'bg-background');
    content = content.replace(/bg-white\/\[?[0-9.]+\]?/g, 'bg-card border border-border shadow-sm');
    
    // Texts
    content = content.replace(/text-foreground/g, 'text-foreground');
    content = content.replace(/text-muted-foreground/g, 'text-muted-foreground');
    content = content.replace(/text-muted-foreground/g, 'text-muted-foreground');
    content = content.replace(/text-muted-foreground text-opacity-80/g, 'text-muted-foreground text-opacity-80');
    content = content.replace(/text-muted-foreground text-opacity-60/g, 'text-muted-foreground text-opacity-60');
    content = content.replace(/text-muted-foreground text-opacity-40/g, 'text-muted-foreground text-opacity-40');
    
    // Borders
    content = content.replace(/border-white\/\[?[0-9.]+\]?/g, 'border-border');
    content = content.replace(/border-border/g, 'border-border');
    content = content.replace(/border-border/g, 'border-border');

    // Hover Backgrounds
    content = content.replace(/hover:bg-white\/\[?[0-9.]+\]?/g, 'hover:bg-accent hover:text-accent-foreground transition-colors');

    // Add bg-card border border-border shadow-sm utility to Tailwind or just use standard
    content = content.replace(/bg-card border border-border shadow-sm/g, 'bg-card border border-border shadow-sm');
    
    if (content !== original) {
        fs.writeFileSync(filepath, content, 'utf-8');
        console.log(`Updated ${filepath}`);
    }
}

function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filepath = path.join(dir, file);
        if (fs.statSync(filepath).isDirectory()) {
            processDirectory(filepath);
        } else if (file.endsWith('.jsx') || file.endsWith('.js')) {
            migrateFile(filepath);
        }
    }
}

processDirectory(__dirname);

File: frontend/src/migrate_theme.py
import os
import re

def migrate_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content

    # Backgrounds
    content = re.sub(r'bg-\[#0a0e1a\]', 'bg-background', content)
    content = re.sub(r'bg-\[#0f1629\]', 'bg-background', content)
    content = re.sub(r'bg-\[#111827\]', 'bg-background', content)
    content = re.sub(r'bg-white/\[?[0-9.]+\]?', 'bg-card', content)
    
    # Texts
    content = re.sub(r'text-white', 'text-foreground', content)
    content = re.sub(r'text-slate-200', 'text-muted-foreground', content)
    content = re.sub(r'text-slate-300', 'text-muted-foreground', content)
    content = re.sub(r'text-slate-400', 'text-muted-foreground', content)
    content = re.sub(r'text-slate-500', 'text-muted-foreground', content)
    content = re.sub(r'text-slate-600', 'text-muted-foreground', content)
    
    # Borders
    content = re.sub(r'border-white/\[?[0-9.]+\]?', 'border-border', content)
    content = re.sub(r'border-slate-800', 'border-border', content)
    content = re.sub(r'border-slate-700', 'border-border', content)

    # Hover Backgrounds
    content = re.sub(r'hover:bg-white/\[?[0-9.]+\]?', 'hover:bg-accent hover:text-accent-foreground', content)
    
    # Opacity effects that make text hard to read in light mode
    # content = re.sub(r'backdrop-blur-xl', 'backdrop-blur-md', content)

    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filepath}")

def main():
    for root, dirs, files in os.walk('.'):
        for file in files:
            if file.endswith('.jsx'):
                migrate_file(os.path.join(root, file))

if __name__ == '__main__':
    main()
